code/modules/clothing/chameleon.dm

```diff
@@ -760,3 +763,167 @@
 /obj/item/clothing/neck/chameleon/broken/Initialize(mapload)
 	. = ..()
 	chameleon_action.emp_randomise(INFINITY)
+
+/obj/item/gun/energy/laser/chameleon
+	var/datum/action/item_action/chameleon/change/chameleon_action
+
+	ammo_type = list(/obj/item/ammo_casing/energy/chameleon)
+	pin = /obj/item/firing_pin
+	automatic_charge_overlays = FALSE
+	can_select = FALSE
+
+	/// The vars copied over to our projectile on fire.
+	var/list/chameleon_projectile_vars
+
+	/// The badmin mode. Makes your projectiles act like the real deal.
+	var/real_hits
+
+/obj/item/gun/energy/laser/chameleon/New()
+	..()
+	chameleon_action = new(src)
+	chameleon_action.chameleon_type = /obj/item/gun
+	chameleon_action.chameleon_name = "Gun"
+	chameleon_action.chameleon_blacklist = typesof(/obj/item/gun/magic, /obj/item/gun/energy/minigun)
+	chameleon_action.initialize_disguises()
+
+/obj/item/gun/energy/laser/chameleon/Initialize(mapload)
+	. = ..()
+	recharge_newshot()
+	get_chameleon_projectile(/obj/item/gun/energy/laser)
+
+/obj/item/gun/energy/laser/chameleon/emp_act(severity)
+	return
+
+/**
+ * Description: Resets the currently loaded chameleon variables, essentially resetting it to brand new.
+ * Arguements: []
+ */
+/obj/item/gun/energy/laser/chameleon/proc/reset_chameleon_vars()
+	chameleon_projectile_vars = list()
+
+	if(chambered)
+		chambered.firing_effect_type = initial(chambered.firing_effect_type)
+
+	fire_sound = initial(fire_sound)
+	burst_size = initial(burst_size)
+	fire_delay = initial(fire_delay)
+	inhand_x_dimension = initial(inhand_x_dimension)
+	inhand_y_dimension = initial(inhand_y_dimension)
+
+	QDEL_NULL(chambered.loaded_projectile)
+	chambered.newshot()
+
+/**
+ * Description: Sets what gun we should be mimicking.
+ * Arguements: [obj/item/gun/gun_to_set (the gun we're trying to mimic), passthrough (whether or not we're setting ammo too)]
+ */
+/obj/item/gun/energy/laser/chameleon/proc/set_chameleon_gun(obj/item/gun/gun_to_set, passthrough = TRUE)
+	if(!istype(gun_to_set))
+		stack_trace("[gun_to_set] is not a valid typepath.")
+		return FALSE
+
+	fire_sound = gun_to_set.fire_sound
+	burst_size = gun_to_set.burst_size
+	fire_delay = gun_to_set.fire_delay
+	inhand_x_dimension = gun_to_set.inhand_x_dimension
+	inhand_y_dimension = gun_to_set.inhand_y_dimension
+
+	if(passthrough)
+		if(istype(gun_to_set, /obj/item/gun/ballistic))
```

Comment:
What happens if it is none of these? Is it going to use the same ammo as before, but look different?